sink/src/main/java/com/spredfast/kafka/connect/s3/sink/S3SinkTask.java

```diff
@@ -23,25 +30,16 @@
 import org.apache.kafka.connect.storage.Converter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import com.amazonaws.services.s3.AmazonS3;
-import com.spredfast.kafka.connect.s3.AlreadyBytesConverter;
-import com.spredfast.kafka.connect.s3.Configure;
-import com.spredfast.kafka.connect.s3.Constants;
-import com.spredfast.kafka.connect.s3.Metrics;
-import com.spredfast.kafka.connect.s3.S3;
-import com.spredfast.kafka.connect.s3.S3RecordFormat;
-import com.spredfast.kafka.connect.s3.S3RecordsWriter;
 
 
 public class S3SinkTask extends SinkTask {
 
 	private static final Logger log = LoggerFactory.getLogger(S3SinkTask.class);
-
-	private Map<String, String> config;
-
 	private final Map<TopicPartition, PartitionWriter> partitions = new LinkedHashMap<>();
-
+	private Map<TopicPartition, OffsetAndMetadata> flushedOffsets = new LinkedHashMap<>();
```

Comment:
Can be made `final`

Reply:
It may not be necessary to track the previously flushed offsets in the task. `WorkerSinkTask` tracks `lastCommittedOffsets`, and it looks like it will combine them with the `taskProvidedOffsets` in `commitOffsets()`.

If that works, then `preCommit()` could be simplified just to filtering out the small partitions and passing the result to the parent method.